socialdistribution/api/views.py

```diff
@@ -37,8 +44,124 @@ def list(self, request, *args, **kwargs):
         return response
 
 
-# class FollowerViewSet(viewsets.GenericViewSet, mixins.RetrieveModelMixin, mixins.ListModelMixin, mixins.UpdateModelMixin, mixins.DestroyModelMixin):
-#     queryset = models.Follower
+class FollowerViewSet(viewsets.GenericViewSet, mixins.RetrieveModelMixin, mixins.ListModelMixin, mixins.UpdateModelMixin, mixins.DestroyModelMixin):
+    queryset = Follower.objects.none()
+    serializer_class = serializers.AuthorSerializer
+
+    def get_follower(self, follower_uuid, context):
+        author = models.Author.objects.filter(uuid = self.kwargs['author_pk']).get()
+        followers = models.Follower.objects.filter(author = author).all()
+
+        follower_qs = followers.filter(follower_uuid = follower_uuid)
+        if follower_qs.exists():
+            # get author serialize
+            author_qs = Author.objects.filter(uuid = follower_uuid)
+            if author_qs.exists():
+                author = author_qs.get()
+                return serializers.AuthorSerializer(author, context = context).data
+            else:
+                # remote author, GET info
+                remote_node_qs = models.UUIDRemoteCache.objects.filter(uuid = follower_uuid)
+                if remote_node_qs.exists():
+                    # TODO: GET it and set it to response.data
+
+                    node = remote_node_qs.get()
+                    url = (yarl.URL(node.host) / node.prefix / 'author' / follower_uuid / '').human_repr()
+
+                    post_response = requests.get(
+                        url,
+                        headers = { 'Accept': 'application/json' },
+                    )
+
+                    post_response.raise_for_status()
+
+                    if post_response.ok:
+                        return serializers.AuthorSerializer(post_response.json(), context = context).data
+
+        return None
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        GET {host_url}/author/{author_uuid}/followers/{foreign_uuid}
+
+        Checks if the provided foreign_uuid is a follower of the user
+        """
+        response = Response()
+        # response = super().retrieve(request, *args, **kwargs)
+        foreign_uuid = kwargs['pk']
+
+        follower = self.get_follower(foreign_uuid, { 'request': request })
+
+        if follower is not None:
+            response.data = follower
+        else:
+            return HttpResponseServerError({ 'message': 'Follower exists locally but author cannot be found' })
```

Comment:
pretty sure you have to throw these error objects instead of returning them, because all I get is "message" when polling the endpoint with a non-existent follower.

But while we're on the subject, do you think a 500 code is the best for an invalid follower? I feel like a 404 would do nicely here.